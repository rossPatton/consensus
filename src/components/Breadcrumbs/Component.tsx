import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
