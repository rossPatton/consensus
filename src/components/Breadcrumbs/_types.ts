export type tProps = {
  crumbs?: ts.breadcrumb[],
};
