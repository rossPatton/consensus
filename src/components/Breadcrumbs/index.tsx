import React, {Fragment, memo} from 'react';
import {Link} from 'react-router-dom';

import {tProps} from './_types';

const Breadcrumbs = memo((props: tProps) => {
  if (!props.crumbs) return null;
  if (!(props.crumbs instanceof Array)) return null;

  return (
    <ul className="text-sm font-bold mb-2 leading-none lsNone flex fxWrap brdB1 mb-2 pb-3">
      {props.crumbs.map((crumb: tCrumb, i) => {
        const isLastItem = i === props.crumbs.length - 1;
        const renderLink = !isLastItem && crumb.to;

        return (
          <Fragment key={i}>
            <li className="mr-1">
              {!renderLink && crumb.display}
              {renderLink && (
                <Link to={`/${crumb.to}`}>
                  {crumb.display}
                </Link>
              )}
            </li>
            {renderLink && (
              <li className="mHide mr-1">/</li>
            )}
          </Fragment>
        );
      })}
    </ul>
  );
});

export default Breadcrumbs;
