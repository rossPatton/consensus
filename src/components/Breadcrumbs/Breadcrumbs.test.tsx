import React from 'react';
import render from 'react-test-renderer';

import Breadcrumbs from '.';

it('renders without crashing', () => {
  render.create(<Breadcrumbs />);
});

it('renders null when no props', () => {
  const component = render.create(<Breadcrumbs />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders breadcrumbs when crumb prop passed in', () => {
  const component = render.create((
    <Breadcrumbs
      crumbs={[{display: 'Test', to: '/test'}]}
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
