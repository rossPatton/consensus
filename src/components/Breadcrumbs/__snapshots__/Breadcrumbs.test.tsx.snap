// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders breadcrumbs when crumb prop passed in 1`] = `
<ul
  className="text-sm text-gray-5 mb-1 flex flex-wrap"
>
  <li
    className="mr-1"
  >
    Test
  </li>
</ul>
`;

exports[`renders null when no props 1`] = `null`;
